### DISCUSSION TITLE
路由器重启后延迟启动、失败重试等问题

### DISCUSSION THREAD

**Discussion by qwerttvv** [messense/aliyundrive-webdav#883] [Q&A]
### 问题描述

路由器定时重启，虽然现在已经是99的优先级，但重启后网络还没有好，程序就起来了，然后报错就挂了

目前我是增加了一个重启的计划任务，定在定时重启路由器之后10分钟再重启程序以便解决问题，例如

    30 4 * * * /etc/init.d/aliyundrive-webdav restart

不知道有没有更好的姿势解决这个问题，比如延迟、或者重试，或者其它什么的

### 重现步骤

重启路由器

### 版本

v2.3.2

### 运行平台

openwrt

### 日志

```
<unknown time> ERROR aliyundrive_webdav::drive: refresh token failed: Request error: error sending request for url (https://aliyundrive-oauth.messense.me/oauth/access_token): error trying to connect: tcp connect error: Network unreachable (os error 101)
Error: get access_token failed
```

**Comment by dalasichen** [comment]
存在同样的问题，X86版。
<unknown time> ERROR aliyundrive_webdav::drive: refresh token failed: Request error: error sending request for url (https://aliyundrive-oauth.messense.me/oauth/access_token): error trying to connect: dns error: failed to lookup address information: Try again
Error: get access_token failed

**Comment by sunfing** [comment]
大佬解决了吗？我基本上每次重启都遇到这个问题“Error: get access_token failed”，按照楼主的计划任务，我让ChatGPT给了我一条系统 crontab 中的计划任务“sleep 90 && /etc/init.d/aliyundrive-webdav restart”，还不知道行不行。

**Reply by qwerttvv** [reply]
阿里云接口限速，我现在都不想用了…
